### ISSUE TITLE
Allow passing ActiveRecord objects to renders_many Component slot

### ISSUE THREAD

**Issue by tenpaiyomi** [ViewComponent/view_component#1547]
## Feature request
I'd like to be able to pass an ActiveRecord object along to a `renders_many` Component slot.  As is right now, the reference to utilizing collections explicitly passes in an array of hashes, and the error given when trying to pass along an ActiveRecord object implies that it is trying to convert the ActiveRecord object into a hash as well instead of utilizing it in a manner that would exist if I passed directly to the component with `with_collection`.

I have a (basic) setup like so:

```
# table_component.rb
class TableComponent < ViewComponent::Base
  renders_many :records, RecordComponent
end
```

```
# table_component.html.erb
<table class='table table-hover'>
  <thead>
    <tr>
      <th scope='col'><%= _('Date') %></th>
      <th scope='col'><%= _('Status') %></th>
      <th scope='col'><%= _('User') %></th>
      <th scope='col'><%= _('Project') %></th>
      <th scope='col'><%= _('Notes') %></th>
  </thead>
  <tbody id='records'>
    <% records.each do |record| %>
      <%= record %>
    <% end %>
  </tbody>
</table>
```

```
#record_component.rb
class RecordComponent < ViewComponent::Base
  def initialize(record:)
    @record = record
  end
end
```

```
# record_component.html.erb
<tr id='<%= helpers.dom_id(@record) %>'>
  <td scope="row"><%= @record.created_at %></td>
  <td scope="row"><%= @record.status %></td>
  <td scope="row" title='<%= @record.user.email %>'><%= @record.user.name %></td>
  <td scope="row"><%= @record.project.name %></td>
  <td scope="row">
    <ul>
      <% @record.notes&.each do |note| %>
        <li><%= note %></li>
      <% end %>
    </ul>
  </td>
</tr>
```

And am calling it like such:

```
#index.html.erb
<%= render(TableComponent.new.with_records(@records)) %>
```

Which winds up giving me this error:

`no implicit conversion of Record into Hash`

### Motivation

Being able to render out a parent component that has many ActiveRecord backed child components would help clean up the code greatly.  I can work around this by passing the records as an argument to the initializer for `TableComponent`, but it feels like it is somewhat contrary to how everything _implies_ that I should be able to work with slots and collections.

**Comment by joelhawksley** [comment]
@tenpaiyomi makes sense to me. Want to write PR?

**Comment by tenpaiyomi** [comment]
> @tenpaiyomi makes sense to me. Want to write a PR?

Yep, I'll see what I can come up with. On a bit of a work crunch right now so will be a bit before I can jump on it.

**Comment by joelhawksley** [comment]
Closed by #1702
